### PR TITLE
fix: Fix parent of new ManagedPolicy created by ConvertInlinePoliciesToManaged

### LINKED ISSUE #733: bug: ConvertInlienPoliciesToManaged breaks dependencies on default policies
### What is the problem?

In some common cases (such as with Pipelines or Lambda Functions), automated conversion of the Policy to a ManagedPolic break existing DependsOn relationships. This causes deployments to fail, as some resources are created before their depended policies. 

### Reproduction Steps

```
    const fn = new Function(stack, 'TestLambda', {
      code: Code.fromInline(`def handler(event, context)\n    print(event)`),
      handler: 'index.handler',
      runtime: Runtime.PYTHON_3_11,
    });
    fn.addToRolePolicy(
      new iam.PolicyStatement({
        actions: ['s3:*'],
        resources: ['*'],
      })
    );
    Aspects.of(app).add(new ConvertInlinePoliciesToManaged());
```

### What did you expect to happen?

The resulting template should maintain the DependsOn referenceing the default policy of the role

### What actually happened?

The DependsOn reference is lost

### cdk-enterprise-iac version

0.0.439

### Language

Typescript, Python

### Other information

_No response_

### PR DESCRIPTION
**Pull request by ryfestag**
This PR updates the scope of the newly-created ManagedPolicy resource to keep the same parent as the deleted resource. A unit test is added to verify that the dependencies aren't lost (previously, the dependencies would only include the Role, not the associated policy)

Fixes #733

### CODE REVIEW

File: test/patches/convertInlinePoliciesToManaged.test.ts [hunk 1]
```diff
@@ -78,6 +78,37 @@ describe('Updating Resource Types', () => {
...
+    );
+
+    Aspects.of(app).add(new ConvertInlinePoliciesToManaged());
+    app.synth();
+    const template = Template.fromStack(stack);
+    console.log((template as any).template.Resources);
+    console.log(
+      (template as any).template.Resources
+        .TestLambdaServiceRoleDefaultPolicy0F2D5E78
+    );
```
Comment (maafk) [inline test/patches/convertInlinePoliciesToManaged.test.ts:101]: Can you remove the console logs in the test?

**Review by maafk — CHANGES_REQUESTED** [review]
Looks great! Just made one change request

File: test/patches/convertInlinePoliciesToManaged.test.ts (hunk 1 above)
Comment (ryfestag) [reply test/patches/convertInlinePoliciesToManaged.test.ts:101]: Ah yea, good call. I'll update

**Review by maafk — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**